SpatialGDK/Source/SpatialGDK/Private/Tests/RPCServiceTest.cpp

```diff
@@ -0,0 +1,568 @@
+			{
+				RPCService.OnEndpointAuthorityGained(EntityId, SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID);
+			}
+
+			if (RPCEndpointType == SERVER_AUTH ||
+				RPCEndpointType == SERVER_AND_CLIENT_AUTH)
+			{
+				RPCService.OnEndpointAuthorityGained(EntityId, SpatialConstants::SERVER_ENDPOINT_COMPONENT_ID);
+			}
+		}
+
+		return RPCService;
+	}
+
+	bool CompareRPCPayload(const SpatialGDK::RPCPayload& Payload1, const SpatialGDK::RPCPayload& Payload2)
+	{
+		return Payload1.Index == Payload2.Index &&
+			   Payload1.Offset == Payload2.Offset &&
+			   Payload1.PayloadData == Payload2.PayloadData;
+	}
+
+	bool CompareSchemaObjectToSendAndPayload(Schema_Object* SchemaObject, const SpatialGDK::RPCPayload& Payload, ERPCType RPCType, uint64 RPCId)
+	{
+		SpatialGDK::RPCRingBufferDescriptor Descriptor = SpatialGDK::RPCRingBufferUtils::GetRingBufferDescriptor(RPCType);
+		Schema_Object* RPCObject = Schema_GetObject(SchemaObject, Descriptor.GetRingBufferElementFieldId(RPCId));
+		return CompareRPCPayload(SpatialGDK::RPCPayload(RPCObject), Payload);
+	}
+
+	bool CompareUpdateToSendAndEntityPayload(SpatialGDK::SpatialRPCService::UpdateToSend& Update, const EntityPayload& EntityPayloadItem, ERPCType RPCType, uint64 RPCId)
+	{
+		return CompareSchemaObjectToSendAndPayload(Schema_GetComponentUpdateFields(Update.Update.schema_type), EntityPayloadItem.Payload, RPCType, RPCId) &&
+			   Update.EntityId == EntityPayloadItem.EntityId;
+	}
+
+	bool CompareComponentDataAndEntityPayload(const Worker_ComponentData& ComponentData, const EntityPayload& EntityPayloadItem, ERPCType RPCType, uint64 RPCId)
+	{
+		return CompareSchemaObjectToSendAndPayload(Schema_GetComponentDataFields(ComponentData.schema_type), EntityPayloadItem.Payload, RPCType, RPCId);
+	}
+
+	const Worker_ComponentData GetComponentDataOnEntityCreationFromRPCService(SpatialGDK::SpatialRPCService& RPCService, Worker_EntityId EntityID, ERPCType RPCType)
+	{
+		Worker_ComponentId ExpectedUpdateComponentId = SpatialGDK::RPCRingBufferUtils::GetRingBufferComponentId(RPCType);
+		TArray<Worker_ComponentData> ComponentDataArray = RPCService.GetRPCComponentsOnEntityCreation(EntityID);
+
+		const Worker_ComponentData* ComponentData = ComponentDataArray.FindByPredicate([ExpectedUpdateComponentId](const Worker_ComponentData& ComponentData) {
+			return ComponentData.component_id == ExpectedUpdateComponentId;
+		});
+
+		if (ComponentData == nullptr)
+		{
+			return {};
+		}
+		return *ComponentData;
+	}
+} // anonymous namespace
+
+RPC_SERVICE_TEST(GIVEN_authority_over_server_endpoint_WHEN_push_client_reliable_rpcs_to_the_service_THEN_rpc_push_result_success)
+{
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, SERVER_AUTH);
+	SpatialGDK::EPushRPCResult Result = RPCService.PushRPC(RPCTestEntityId_1, ERPCType::ClientReliable, SimplePayload);
+	bool bTestPassed = (Result == SpatialGDK::EPushRPCResult::Success);
+	TestTrue("Push RPC returned expected results", bTestPassed);
```

Comment:
```suggestion
	TestTrue("Push RPC returned expected results", (Result == SpatialGDK::EPushRPCResult::Success));
```

Reply:
Can fold these lines into one on all the tests now, since we're just returning true.

Reply:
Done